cmd2/cmd2.py

```diff
@@ -381,6 +399,229 @@ def __init__(self, completekey: str = 'tab', stdin=None, stdout=None, *,
         # If False, then complete() will sort the matches using self.default_sort_key before they are displayed.
         self.matches_sorted = False
 
+    def _autoload_commands(self) -> None:
+        """Load modular command definitions."""
+        # Search for all subclasses of CommandSet, instantiate them if they weren't provided in the constructor
+        all_commandset_defs = CommandSet.__subclasses__()
+        existing_commandset_types = [type(command_set) for command_set in self._installed_command_sets]
+        for cmdset_type in all_commandset_defs:
+            init_sig = inspect.signature(cmdset_type.__init__)
+            if cmdset_type in existing_commandset_types or \
+                    len(init_sig.parameters) != 1 or \
+                    'self' not in init_sig.parameters:
+                continue
```

Comment:
Recommend adding a unit test to exercise the case where this `continue` is reached